draft-bbs-signatures.md

```diff
@@ -475,33 +483,25 @@ Procedure:
 
 2. (i1, i2, ..., iR) = RIndxs
 
-3. (A', Abar, d, C1, e^, r2^, C2, r3^, s^, (m^\[i1\],...,m^\[iR\])) = spk
-
-4. (w, h0, h\[1\],...,h\[L\]) = PK
-
-5. if A' == 1 return INVALID
-
-6. X1 = e(A', w)
-
-7. X2 = e(Abar, P2)
-
-8. if X1 != X2 return INVALID
+3. (A', Abar, D, c, e^, r2^, r3^, s^, (m^[i1],...,m^[iR])) = spk
 
-9. c = H(Abar || A' || h0 || C1 || d || h0 || h\[i1\] || ... || h\[iR\] || C2 || pm)
+4. (W, H0, H[1],...,H[L]) = PK
 
-10. T1 = Abar - d
+5. C1 = (Abar - D) *c + A' * e^ + H0 * r2^ 
 
-11. T2 = P1 + h\[i1\] \* Rmsg\[1\] + ... + h\[iR\] \* Rmsg\[R\]
+6. T = P1 + H[i1] * Rmsg[i1] + ... H[iR] * Rmsg[R]
 
-12. Y1 = A' \* e^ + h0 \* r2^ + T1 \* c
+7. C2 = T * c - D * r3^ + H0 * s^ + H[i1] * m^[i1] + ... + H[iR] * m^[iR]
+ 
+8. cv = HASH(PK || Abar || A' || D || C1 || C2 || pm)
 
-13. Y2 = d \* r3^ + h0 \* s^ + h\[i1\] \* m^\[i1\] + ... + h\[iR\] \* m^\[iR\] - T2 \* c
+9. if c != cv return INVALID
 
-14. if C1 != Y1 return INVALID
+10. if A' == 1 return INVALID
 
-15. if C2 != Y2 return INVALID
+11. if e(A', W) * e(Abar, -P2) != 1 return INVALID
```

Comment:
Why not just check for `e(A', W) == e(Abar, P2)`??

Reply:
Because mathematically it’s quicker. You can compute what I put with a single pairing versus two pairings. Most folks don’t know that. It’s also preferable this way.

Reply:
Got it. Maye be we should replace it with `e(A' + Abar, W - P2) != 1` then??

Reply:
Well its can be written that way but if you were to actually compute it that way it might not work.

Reply:
If that's the case then i don't see the difference with checking for equality. Don't you still compute 2 pairings??

Reply:
I can give you a code sample to illustrate

Reply:
Please do!! thanks!!